File: Shell_Retail-main/CJ_Shopping/ecomm/Product.java
package ecomm;
import java.util.*;
public abstract class Product {
	public abstract void setVals(String s,String i,Float p,int n);
	// common queries to get category, unique name, price, and number available
	public abstract Globals.Category getCategory();
	public abstract String getName();
	public abstract String getProductID();
	public abstract float getPrice();
	public abstract int getQuantity();
	public abstract void setQuantity(int q);
}

File: Shell_Retail-main/CJ_Shopping/product.cpp
#include <bits/stdc++.h>
using namespace std;
#include "product.h"
product :: product( string cat, string ne, int pr, int num)
{
    category = cat;
    name = ne;
    number = num;
    price = pr;
}

string product :: get_name()    //returns name.
{ 
    return name;
}

int product :: get_price()   //returns price.
{
    return price;
}

int product :: get_number()   //returns number.
{
    return number;
}
string product :: get_category(){
    return category;
}
void product :: Print()    //To print the desired output.
{
    cout << category << " " << name << " " << number << " " << price << endl;
}